context/_coroutine.c

```diff
@@ -80,8 +80,24 @@ static void *coroutine_wrapper(void *action_, void *arg_)
     PyThreadState *thread_state = PyThreadState_GET();
 
     /* New coroutine gets a brand new Python interpreter stack frame. */
+#if PY_VERSION_HEX >= 0x30B0000
     PyThreadState *new_threadstate = PyThreadState_New(thread_state->interp);
-    PyThreadState_Swap(new_threadstate);
+    thread_state = PyThreadState_Swap(new_threadstate);
```

Comment:
It's a little odd that we're now assigning `thread_state`, but weren't before; I guess this variable isn't used subsequently.